.github/workflows/gcc-build.yml

```diff
@@ -8,36 +8,50 @@ jobs:
   build:
     runs-on: ubuntu-22.04
     steps:
-    - uses: actions/checkout@v4
-    - name: Set reusable strings
-      id: strings
-      shell: bash
-      run: |
-        echo "build-output-dir=${{ github.workspace }}/bin" >> "$GITHUB_OUTPUT"
-    - name: Dependencies
-      run: |
-        sudo apt-get update && sudo apt-get install -yq libboost-all-dev g++-11
-        sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-11 100 --slave /usr/bin/g++ g++ /usr/bin/g++-11
-    - name: Setup
-      env:
-        CFLAGS: -Werror
-        CXXFLAGS: -Werror
-        CMAKE_BUILD_TYPE: Debug
-      run: >
-        cmake -S ${{ github.workspace }} -B ${{ steps.strings.outputs.build-output-dir }}
-        -DWITH_WARNINGS=1 -DWITH_COREDEBUG=0 -DUSE_COREPCH=1 -DUSE_SCRIPTPCH=1 -DTOOLS=1 -DSCRIPTS=dynamic -DSERVERS=1 -DNOJEM=0
-        -DCMAKE_C_FLAGS_DEBUG="-DNDEBUG" -DCMAKE_CXX_FLAGS_DEBUG="-DNDEBUG"
-        -DCMAKE_INSTALL_PREFIX=check_install -DBUILD_TESTING=1
-    - name: Build
-      run: |
-        cd bin
-        make -j 4 -k && make install
-    - name: Unit tests
-      run: |
-        cd bin
-        make test
-    - name: Check executables
-      run: |
-        cd ${{ github.workspace }}/check_install/bin
-        ./bnetserver --version
-        ./worldserver --version
+      - uses: actions/checkout@v4
+      - name: Set reusable strings
+        id: strings
+        shell: bash
+        run: |
+          echo "build-output-dir=${{ github.workspace }}/bin" >> "$GITHUB_OUTPUT"
+      - name: Dependencies
+        run: |
+          sudo apt-get update && sudo apt-get install -yq libboost-all-dev g++-11
+          sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-11 100 --slave /usr/bin/g++ g++ /usr/bin/g++-11
+      - name: Setup
+        env:
+          CFLAGS: -Werror
+          CXXFLAGS: -Werror
+          CMAKE_BUILD_TYPE: Debug
+        run: >
+          cmake -S ${{ github.workspace }} -B ${{ steps.strings.outputs.build-output-dir }}
+          -DWITH_WARNINGS=1 -DWITH_COREDEBUG=0 -DUSE_COREPCH=1 -DUSE_SCRIPTPCH=1 -DTOOLS=1 -DSCRIPTS=dynamic -DSERVERS=1 -DNOJEM=0
+          -DCMAKE_C_FLAGS_DEBUG="-DNDEBUG" -DCMAKE_CXX_FLAGS_DEBUG="-DNDEBUG"
+          -DCMAKE_INSTALL_PREFIX=check_install -DBUILD_TESTING=1
+      - name: Build
+        run: |
+          cd bin
+          make -j 4 -k && make install
+      - name: Unit tests
+        run: |
+          cd bin
+          make test
+      - name: Check executables
+        run: |
+          cd ${{ github.workspace }}/check_install/bin
+          ./bnetserver --version
+          ./worldserver --version
+
+      - name: Archive
+        if: github.ref == 'refs/heads/master'
```

Comment:
is this check similar to https://github.com/TrinityCore/TrinityCore/blob/b1d8061a0f848241fe70ab0d718e829836cb5601/.circleci/config.yml#L96C21-L96C22 or ? As in, "master" is the name of the branch and multiple branches should be supported ?